yt_dlp/extractor/instagram.py

```diff
@@ -356,113 +357,197 @@ def _extract_embed_url(webpage):
-            'text': traverse_obj(comment_dict, ('node', 'text')),
-            'timestamp': traverse_obj(comment_dict, ('node', 'created_at'), expected_type=int_or_none),
-        } for comment_dict in comment_data] if comment_data else None
-
-        display_resources = (
-            media.get('display_resources')
-            or [{'src': media.get(key)} for key in ('display_src', 'display_url')]
-            or [{'src': self._og_search_thumbnail(webpage)}])
-        thumbnails = [{
-            'url': thumbnail['src'],
-            'width': thumbnail.get('config_width'),
-            'height': thumbnail.get('config_height'),
-        } for thumbnail in display_resources if thumbnail.get('src')]
-
         return {
-            'id': video_id,
-            'formats': formats,
-            'title': media.get('title') or 'Video by %s' % username,
-            'description': description,
-            'duration': float_or_none(media.get('video_duration')),
-            'timestamp': traverse_obj(media, 'taken_at_timestamp', 'date', expected_type=int_or_none),
-            'uploader_id': traverse_obj(media, ('owner', 'id')),
-            'uploader': traverse_obj(media, ('owner', 'full_name')),
-            'channel': username,
-            'like_count': self._get_count(media, 'likes', 'preview_like') or str_to_int(self._search_regex(
-                r'data-log-event="likeCountClick"[^>]*>[^\d]*([\d,\.]+)', webpage, 'like count', fatal=False)),
-            'comment_count': self._get_count(media, 'comments', 'preview_comment', 'to_comment', 'to_parent_comment'),
-            'comments': comments,
-            'thumbnails': thumbnails,
+            'id':
+            video_id,
+            'formats':
+            formats,
+            'title':
+            f'Video by {username}',
+            'description':
+            traverse_obj(info, ('caption', 'text'), expected_type=str),
+            'duration':
+            float_or_none(info.get('video_duration')),
+            'timestamp':
+            traverse_obj(info, 'taken_at', expected_type=int_or_none),
+            'uploader_id':
+            str(traverse_obj(info, ('user', 'pk'), expected_type=int)),
+            'uploader':
+            traverse_obj(info, ('user', 'full_name'), expected_type=str),
+            'channel':
+            username,
+            'like_count':
+            traverse_obj(info, 'like_count', expected_type=int_or_none),
+            'comment_count':
+            traverse_obj(info, 'comment_count', expected_type=int_or_none),
+            'thumbnails': [{
+                'url': thumbnail['url'],
+                'width': thumbnail.get('width'),
+                'height': thumbnail.get('height'),
+            } for thumbnail in info.get('image_versions2', {}).get(
+                'candidates', []) if thumbnail.get('url')],
             'http_headers': {
                 'Referer': 'https://www.instagram.com/',
             }
         }
```

Comment:
I hope that it will work, but i just don't know how to implement it with `_extract_product(info)`. My code is a base fix and can be optimised so i need your help to make it upto standards.

Reply:
the entire code i highlighted can be replaced with that one line